### ISSUE TITLE
New instrument; Power meter

### ISSUE THREAD

**Issue by cweitat** [fossasia/pslab-android#1392]
**Actual Behaviour**
Currently there is no instrument that is able to measure watts.

**Expected Behaviour**
There should be an instrument to measure the watts
Can also integrate watts to multimeter, oscilloscope , etc

**Would you like to work on the issue?**

Anyone.

**Comment by naman653** [comment]
I would like to work on the implementation of this feature. Don't know much about the codebase of this project but I'm willing to help and contribute.

**Comment by ho-dor** [comment]
@cweitat i would like to work on this

**Comment by riyasingh1004** [comment]
Hi, I would like to work on this issue.

**Comment by neel1998** [comment]
@cweitat  should this be included in multimeter it self or there should be a separate screen ? May be we can combine #1426 and this instrument together in one screen.

**Comment by cweitat** [comment]
@neel1998 this was mentioned few meetings ago to include within multimeter.

**Comment by neel1998** [comment]
@cweitat  yes, but multimeter is already very crowded that's why I asked. If we are going to read watts for the same pins as there are currently in multimeter, may be we can add a separate block in display which shows watt as well.

**Comment by neel1998** [comment]
@cweitat for which pins exactly we plan to read power?

**Comment by cweitat** [comment]
Derive from voltage

**Comment by neel1998** [comment]
@cweitat where can I find which resistor is connected to which pins on PSLab internally. So I can calculate the power using P = V^2/R. I looked into the PSLab data sheet but resistor information is not given there.

**Comment by cweitat** [comment]
@CloudyPadmal

**Comment by CloudyPadmal** [comment]
Well, I think there's no direct way to measure power. Will have to connect a known resistor to measure current and then measure the voltage drop across the equipment to calculate I^2R. Not sure about the practicality of this method though. :thinking:

**Comment by CloudyPadmal** [comment]
Hardware based solution would be better. Moving this to pslab-hardware repo
